vitrina/uapi/views/views.py

```diff
@@ -60,6 +60,11 @@ class DatasetViewSet(UAPIExceptionHandlerMixin, viewsets.ModelViewSet):
     def dataset_metadata_id(self) -> int:
         return ContentType.objects.get_for_model(Dataset).id
 
+    @cached_property
+    def dcat_resource_subclass(self) -> DCATResourceSubclass:
+        subclass_name = self.request.data.get("subclass", DCATResourceSubclass.DATASET)
+        return DCATResourceSubclass.objects.get(name=subclass_name)
```

Comment:
get_object_or_404, as filter param is retrieved from the request and can be incorrect

Reply:
Good idea, fixed:) [e44551b](https://github.com/atviriduomenys/katalogas/pull/1829/commits/e44551b598b5225a5ebad7619e642110ffe27b41)